### ISSUE TITLE
Owncloud Client Mac does not save password

### ISSUE THREAD

**Issue by ohnemich** [owncloud/client#5750]
Hi,
my issue seems to be related to apples keychain access. After an initial (successful) setup of an account in the Mac OS X client, closing owncloud client and restarting it, I am not able to login any more. See more detailed "steps to reproduce". Would appreciate any hint, I do not know how to deal with this issue. Thanks!

### Expected behaviour
Owncloud desktop file client should login with the stored credentials/password.

### Actual behaviour
Owncloud desktop client does not login.

### Steps to reproduce
1. Start Owncloud Client
2. Set up an owncloud account which is working (sync works well)
3. Close Owncloud Client (Client is not running in the background)
4. Start Owncloud Client
5. OC Client Asking for Password.
6. Using pre-stored password from keychain access (*) does not work, manual enter of password neither (entered via copy and paste, password tested with web login).
7. Owncloud is not connected, not able to sync
8. To make Owncloud work again, it is necessary to delete the account and set it up again. Then it is working, until owncloud client is closed and restarted, there it starts with 4.

*(Credentials are stored after an initial setup of the account or after deleting and setting up account again - if I login after being at step 6, he deletes the owncloud entries for clientKeyPEM and clientCertificatePEM and the password and always generates new keychain access entries for those two owncloud entries (not for the password, perhaps as it is denied as "not correct") at every login attempt)

### Server configuration
Operating system: unknown

Web server: unknown

Database:unknown

PHP version:unknown

ownCloud version: 9.1.4

Storage backend (external storage):unknown

### Client configuration
Client version:2.3.1 build 4139

Operating system:Mac OS X Sierra 10.12.4

OS language:German

Qt version used by client package (Linux only, see also Settings dialog):

Client package (From ownCloud or distro) (Linux only):

Installation path of client:

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

```Template for output < 10 lines```

1. Client logfile (can attach further logs, if necessary)
05-07 18:57:00:768 8770 OCC::ConnectionValidator::slotStatusFound: ** Application: ownCloud found:  QUrl("URL")  with version  "9.1.4" ( "9.1.4.2" )
OCC::AccountState::slotConnectionValidatorResult: AccountState connection status change:  "Undefined" -> "Credentials Wrong"
05-07 18:57:00:769 8770 OCC::HttpCredentials::fetchFromKeychain: -------- -----> QSslCertificate("", "", "1B2M2Y8AsgTpgAmY7PhCfg==", (), (), QMap(), QDateTime( Qt::TimeSpec(LocalTime)), QDateTime( Qt::TimeSpec(LocalTime))) QSslKey(PrivateKey, OPAQUE, -1)
05-07 18:57:00:770 8770 OCC::AccountState::setState: AccountState state change:  "Getrennt" -> "Konfigurationsfehler"

**Comment by SamuAlfageme** [comment]
Hey @ohnemich thanks for reporting! Are you positive that the client is creating a new password entry **on every login**: erasing the account does not remove the keychain data relative to the account's password, just logout does.

![screenshot_2017-05-08_09_21_25](https://cloud.githubusercontent.com/assets/2644445/25795037/d0219a56-33d3-11e7-9c32-13f8b61e1ed5.png)

This makes me think of an error accessing the keychain (I experienced something similar on when I was not properly linking the keychain-wrapper lib in https://github.com/owncloud/client/issues/5430). 

If you can, remove all the ownCloud entries on the Keychain Access app to verify the 3 new are created on account login, and additionally, the password entry is deleted on logout, that could be a symptom of this. Also, can you check if `ownCloud.app` appears in the `Access Control` tab of the keychain password item?

<p align="center">
  <img src="https://cloud.githubusercontent.com/assets/2644445/25794960/6a69eee8-33d3-11e7-9143-245093ed769e.png"/>
</p>

**Comment by guruz** [comment]
Linking https://central.owncloud.org/t/owncloud-client-mac-does-not-save-password/7508

**Comment by ohnemich** [comment]
Hi @SamuAlfageme thanks for your reply. Unfortunately, I faced several other issues with my new mac (which i set up on saturday evening) and as I could not solve them and need a running system, I turned back to ubuntu (facing another problem with the owncloud client) for the moment. I will try to answer your hints the best way I can remember. The device is turned back to the shop and can not be used for more tries, sorry.

Check if "_ownCloud.app appears in the Access Control tab_" - I checked this, owncloud.app was set correct.

I do not understand your explanation in the first paragraph completely. After initial successful setup, I had three entries in the keychain access. I closed owncloud client (I did not try to logout seperately!), and I restarted it, still seeing three entries. After the restart, he did not login with my password and generated new entries (now only two entries), everytime I entered my password and klicked ok.

I tried to delete he entries in the keychain access (a) after initial setup and b) after some login attempts) by myself, because I thought, he could generate new ones without referring to defective old ones - but it did not help neither,

Would love to help you anymore, if I can and you suspect owncloud client to be the cause.
